src/main/java/de/tum/in/www1/artemis/service/metis/PostingService.java

```diff
@@ -141,8 +142,19 @@ Course preCheckUserAndCourse(User user, Long courseId) {
         authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.STUDENT, course, user);
 
         // check if the course has posts enabled
-        if (!course.getPostsEnabled()) {
-            throw new BadRequestAlertException("Postings are not enabled for this course", getEntityName(), "400", true);
+        if (!courseRepository.isCommunicationEnabled(courseId)) {
+            throw new BadRequestAlertException("Communication feature is not enabled for this course", getEntityName(), "400", true);
+        }
+        return course;
+    }
+
+    Course preCheckUserAndCourseForConversation(User user, Conversation conversation, Long courseId) {
```

Comment:
```suggestion
    Course getCourseIfMessagingPermittedElseThrow(User user, Conversation conversation, Long courseId) {
```
For me personally the naming does not really fit. I’d not expect a ‘check‘ method to also fetch something.